packages/protocol/contracts/governance/BondedDeposits.sol

```diff
@@ -562,22 +564,23 @@ contract BondedDeposits is IBondedDeposits, ReentrancyGuard, Initializable, Usin
   function _redeemRewards(address _account) private returns (uint256) {
     Account storage account = accounts[_account];
     uint256 rewardBlockNumber = block.number.sub(1);
-    FractionUtil.Fraction storage previousCumulativeRewardWeight = cumulativeRewardWeights[
+    int256 previousCumulativeRewardWeight = cumulativeRewardWeights[
       account.rewardsLastRedeemed
     ];
-    FractionUtil.Fraction storage cumulativeRewardWeight = cumulativeRewardWeights[
+    int256 cumulativeRewardWeight = cumulativeRewardWeights[
       rewardBlockNumber
     ];
     // We should never get here except in testing, where cumulativeRewardWeight will not be set.
-    if (!previousCumulativeRewardWeight.exists() || !cumulativeRewardWeight.exists()) {
+    if (previousCumulativeRewardWeight == 0 || cumulativeRewardWeight == 0) {
       return 0;
     }
 
-    FractionUtil.Fraction memory rewardWeight = cumulativeRewardWeight.sub(
+    int256 rewardWeight = cumulativeRewardWeight.subtract(
       previousCumulativeRewardWeight
     );
-    require(rewardWeight.exists(), "Rewards weight does not exist");
-    uint256 value = rewardWeight.mul(account.weight);
+    require(rewardWeight != 0, "Rewards weight does not exist");
+    uint256 value =
+      uint256(rewardWeight.multiply(toFixed(account.weight)).fromFixed());
```

Comment:
pretty sure this is a bug.
casting a `int` to `uint` will interpret some negative numbers as really big one uints

Reply:
If we make sure or operands are positive, this shouldn't happen. Then again, might want to add a >= check just to be safe before conversion.

Reply:
I think casting is dangerous. What about adding a `toUint` in `UsingFixidity` contract?
And we can do the check there